packages/next/build/webpack-config.ts

```diff
@@ -175,6 +176,110 @@ export default async function getBaseWebpackConfig(
 
   const devtool = dev ? 'cheap-module-source-map' : false
 
+  // Contains various versions of the Webpack SplitChunksPlugin used in different build types
+  const splitChunksConfigs: {
+    [propName: string]: webpack.Options.SplitChunksOptions
+  } = {
+    dev: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+      },
+    },
+    selective: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prod: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        commons: {
+          name: 'commons',
+          chunks: 'all',
+          minChunks: totalPages > 2 ? totalPages * 0.5 : 2,
+        },
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prodGranular: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        framework: {
+          name: 'framework',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom|scheduler|prop-types)[\\/]/,
+          priority: 40,
+        },
+        lib: {
+          test(module: { size: Function }): boolean {
+            return module.size() > 160000
+          },
+          name(module: { identifier: Function }): string {
+            let nodeModuleNameResults = /(?:^|\/)node_modules\/(.*)/.exec(
+              module.identifier()
+            )
+            return nodeModuleNameResults
+              ? nodeModuleNameResults[1]
+              : module.identifier()
+          },
+          priority: 30,
+          minChunks: 1,
+          reuseExistingChunk: true,
+        },
+        commons: {
+          name: 'commons',
+          chunks: 'all',
+          minChunks: totalPages,
+          priority: 20,
+        },
+        shared: {
+          name(module, chunks) {
+            return crypto
+              .createHash('sha1')
```

Comment:
Just wanted to check if it'd be worth using a cheaper hashing algo like MD5 here.

Reply:
From my (admittedly pretty cursory) research on this it looks like SHA-1 may be faster in Node? https://medium.com/@chris_72272/what-is-the-fastest-node-js-hashing-algorithm-c15c1a0e164e